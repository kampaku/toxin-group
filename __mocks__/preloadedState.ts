const currentRooms = [
  {
    accessibility: { wideCorridor: true, assistant: false },
    additionalFacilities: {
      breakfast: true,
      crib: false,
      feedingChair: false,
      shampoo: false,
      tv: false,
      writingDesk: false,
    },
    freeDays: {
      from: '2022-10-11T06:17:02.467Z',
      to: '2022-10-18T00:20:15.895Z',
    },
    guests: { kids: 2, baby: 0, adults: 4 },
    id: 10,
    imgSrc: [
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
    ],
    number: 111,
    price: 11423,
    rating: 5,
    reviewsAmount: 99,
    roomAmenities: { bedrooms: 2, beds: 4, bathrooms: 0 },
    rules: { allowSmoke: false, allowGuests: true, allowPets: true },
    type: 'Люкс',
  },
  {
    accessibility: { wideCorridor: true, assistant: false },
    additionalFacilities: {
      breakfast: true,
      crib: false,
      feedingChair: false,
      shampoo: false,
      tv: false,
      writingDesk: false,
    },
    freeDays: {
      from: '2022-10-11T06:17:02.467Z',
      to: '2022-10-18T00:20:15.895Z',
    },
    guests: { kids: 2, baby: 0, adults: 4 },
    id: 20,
    imgSrc: [
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
    ],
    number: 222,
    price: 11423,
    rating: 5,
    reviewsAmount: 99,
    roomAmenities: { bedrooms: 2, beds: 4, bathrooms: 0 },
    rules: { allowSmoke: true, allowGuests: true, allowPets: false },
    type: 'Люкс',
  },
];

const filters = {
  accessibility: {
    wideCorridor: false,
    assistant: false,
  },
  additionalFacilities: {
    breakfast: false,
    crib: false,
    feedingChair: false,
    shampoo: false,
    tv: false,
    writingDesk: false,
  },
  freeDays: {
    from: null,
    to: null,
  },
  guests: { kids: 0, adults: 0, baby: 0 },
  price: { from: 5000, to: 10000 },
  roomAmenities: { bedrooms: 0, beds: 0, bathrooms: 0 },
  rules: { allowGuests: false, allowSmoke: false, allowPets: false },
};

const user = {
  id: '',
  name: '',
  surname: '',
  email: '',
  birthday: '',
  gender: '',
  getSpecOffers: false,
  likes: [],
};
  
const booking = {
  data: [
    {
      date: { from: '21.08.2022', to: '31.08.2022' },
      guests: { adults: 2, baby: 2, kids: 2 },
      totalCost: 2000,
      imgSrc: [''],
      number: 123,
      price: 20000,
      rating: 3,
      reviewsAmount: 10,
      type: '',
    },
  ],
  status: 'fulfilled',
};

export const preloadedState = {
  rooms: { current: currentRooms },
  searchFilters: filters,
  user,
  booking
};
