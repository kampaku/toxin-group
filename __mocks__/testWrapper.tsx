import React, { PropsWithChildren } from 'react';
import { configureStore, PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { NextRouter } from 'next/router';
import { RouterContext } from 'next/dist/shared/lib/router-context';
import { render, RenderOptions } from '@testing-library/react';

import { createMockRouter } from './createMockRouter';

import { combinedReducer, RootState } from '../src/redux/store';

const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: combinedReducer,
    preloadedState,
  });
};

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: ReturnType<typeof setupStore>;
}

const renderWithProviders = (
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
  router: Partial<NextRouter> = {}
) => {
  const Wrapper = ({ children }: PropsWithChildren<{}>): JSX.Element => {
    return (
      <Provider store={store}>
        <RouterContext.Provider value={createMockRouter(router)}>
          {children}
        </RouterContext.Provider>
      </Provider>
    );
  };
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders };
