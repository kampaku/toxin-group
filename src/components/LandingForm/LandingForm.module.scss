@import 'scss/utils/vars';
@import 'scss/utils/font-template';

.landingForm {
  padding: 40px 30px 30px;
  border-radius: 4px;
  box-shadow: 0 0 25px rgb(0 0 0 / 20%);
  background: #fff;
  border: 1px solid rgb(0 0 0 / 12%);
  max-width: 380px;
  width: 100%;
  box-sizing: border-box;

  @media (max-width: 370px) {
    padding: 25px 15px 20px;
  }

  &__title {
    @extend %sub-section-title;

    margin: 0;
  }

  &__calendarWrapper {
    margin: 21px 0;
  }

  &__dropdownWrapper {
    margin-bottom: 31px;

    &Text {
      @extend %label-text;

      text-transform: uppercase;
    }
  }
}
