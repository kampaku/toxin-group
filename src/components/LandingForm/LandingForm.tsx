import { FC } from 'react';
import { useRouter } from 'next/router';
import { useTranslation } from 'next-i18next';

import { Button, Calendar, ConfigurationDropdown } from 'components';
import { useAppDispatch } from 'hooks/hooks';
import { updateFreeDays, updateGuests } from 'redux/slices/searchFiltersSlice';

import styles from './LandingForm.module.scss';

type Props = {
  freeDays: FreeDays;
  guests: Guests;
};

const LandingForm: FC<Props> = ({ freeDays, guests }) => {
  const dispatch = useAppDispatch();
  const router = useRouter();
  const { t } = useTranslation('landing-form');

  const handleSelectedDaysChange = (date: Date[]) => {
    const selectedDays = {
      from: date[0] ? date[0].toISOString() : null,
      to: date[1] ? date[1].toISOString() : null,
    };
    dispatch(updateFreeDays(selectedDays));
  };

  const handleGuestsChange = (content: any) => {
    dispatch(updateGuests(content));
  };

  const onClick = () => {
    router.push('/room-search');
  };

  return (
    <form className={styles.landingForm} onSubmit={(e) => e.preventDefault()}>
      <p className={styles.landingForm__title}>{t('title')}</p>
      <div className={styles.landingForm__calendarWrapper}>
        <Calendar
          mode="twin"
          onSelectedDate={handleSelectedDaysChange}
          selectedDates={freeDays}
        />
      </div>
      <div className={styles.landingForm__dropdownWrapper}>
        <ConfigurationDropdown
          title="guestsTitle"
          type="twoAndOne"
          onChange={handleGuestsChange}
          initialContent={guests}
        />
      </div>
      <div>
        <Button onClick={onClick} type="submit">
          {t('button')}
        </Button>
      </div>
    </form>
  );
};

export { LandingForm };
