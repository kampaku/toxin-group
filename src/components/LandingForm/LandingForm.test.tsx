import { renderWithProviders } from '../../../__mocks__/testWrapper';
import { LandingForm } from './LandingForm';

describe('Sign-in page', () => {
  test('Landing form snapshot', () => {
    const form = renderWithProviders(
      <LandingForm
        freeDays={{
          from: null,
          to: null,
        }}
        guests={{
          adults: 0,
          baby: 0,
          kids: 0,
        }}
      />,
      { preloadedState: {} },
      { asPath: '/' }
    ).container;

    expect(form).toMatchSnapshot();
  });
});
