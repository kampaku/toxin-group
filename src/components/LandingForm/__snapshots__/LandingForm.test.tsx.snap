// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign-in page Landing form snapshot 1`] = `
<div>
  <form
    class="landingForm"
  >
    <p
      class="landingForm__title"
    >
      title
    </p>
    <div
      class="landingForm__calendarWrapper"
    >
      <div
        class="calendar"
      >
        <div
          class="wrapper"
        >
          <label
            class="label"
          >
            labelFirstTwin
            <input
              class="input"
              placeholder="placeholderTwin"
              readonly=""
              type="text"
            />
            <svg
              class="expandIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </label>
          <label
            class="label"
          >
            labelSecond
            <input
              class="input"
              placeholder="placeholderTwin"
              readonly=""
              type="text"
            />
            <svg
              class="expandIcon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </label>
        </div>
      </div>
    </div>
    <div
      class="landingForm__dropdownWrapper"
    >
      <div
        class="roomDropdown"
      >
        <p
          class="roomDropdown__title"
        >
          guestsTitle
        </p>
        <div
          class="roomDropdownInput"
          role="button"
          tabindex="0"
        >
          <div
            class=""
          >
            <input
              class="input input_isReadonly"
              name=""
              placeholder="Email"
              readonly=""
              type="text"
              value="defaultGuests"
            />
          </div>
          <span
            class="roomDropdownInput__iconWrapper"
          >
            <svg
              class="roomDropdownInput__icon"
              fill="none"
              height="20px"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polyline
                points="6 9 12 15 18 9"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
    <div>
      <button
        class="button button_type_submit"
        type="submit"
      >
        <span
          class="button__icon"
        >
          <svg
            fill="currentColor"
            height="24px"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </span>
        <span
          class="button_type_signInText"
        >
          button
        </span>
      </button>
    </div>
  </form>
</div>
`;
